orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/job/RunJobStage.java

```diff
@@ -56,6 +76,43 @@ public void taskGraph(Stage stage, TaskNode.Builder builder) {
     }
   }
 
+  @Override
+  public Result cancel(Stage stage) {
+    log.info(
+        "Canceling run job stage {} for executionId {}",
+        stage.getId(),
+        stage.getExecution().getId());
+    Map<String, Object> destroyContext = new HashMap<>();
+
+    try {
+      RunJobStageContext context =
+          objectMapper.convertValue(stage.getContext(), RunJobStageContext.class);
+      String jobId = context.getJobStatus().getId();
+
+      if (context.getCloudProvider().equalsIgnoreCase("titus")) {
+        destroyContext.put("jobId", jobId);
+      } else {
+        destroyContext.put("jobName", context.getJobStatus().getName());
+      }
+
+      destroyContext.put("cloudProvider", context.getCloudProvider());
+      destroyContext.put("region", context.getJobStatus().getRegion());
+      destroyContext.put("credentials", context.getCredentials());
+
+      stage.setContext(destroyContext);
+
+      destroyJobTask.execute(stage);
+    } catch (Exception e) {
+      log.error(
+          String.format(
```

Comment:
nit: some weird parens here. also, for my education, is `RunJob` titus specific? if not, probably worth making this log message more generic

Reply:
Oops good call. Run job is for titus and kubernetes, so I'll make this generic. I'll also fix the parens